zkevm-circuits/src/witness/block.rs

```diff
@@ -163,8 +227,18 @@ impl BlockContext {
                 [
                     Value::known(F::from(BlockContextFieldTag::Difficulty as u64)),
                     Value::known(F::ZERO),
-                    randomness
-                        .map(|randomness| rlc::value(&self.difficulty.to_le_bytes(), randomness)),
+                    {
+                        if self.difficulty.is_zero() {
```

Comment:
eip-4399

Reply:
I think for Taiko you can just set the value to `mix_hash`, because we're not interested in legacy blocks. But if other code still depends on the old difficulty value then this makes a lot of sense.

Reply:
Make sense, I will check when only use the mix_hash

Reply:
The newest commit has removed the unnecessary difficulty. And all tests passed. Please take another look @Brechtpd